### PR TITLE
UAR-1586 Add property land owner nocs for Gov Public Authority BO

### PR DESCRIPTION
**Pull request by mwestacott**
### JIRA link

https://companieshouse.atlassian.net/browse/UAR-1586

### Change description

* Added new natures of controls to the BO 'gov' page on Registration journey
* Amended NOC validation for the BO gov
* Changed some HTML checks as new NOCs now apply to all types of BOs
* Can be switched on/off with feature flag FEATURE_FLAG_ENABLE_PROPERTY_OR_LAND_OWNER_NOC_30082024
* Unit tests added and amended

### Work checklist

- [X] Tests added where applicable
- [ ] UI changes meet accessibility criteria

### Merge instructions

We are committed to keeping commit history clean, consistent and linear. To achieve this, this commit should be structured as follows:

```
<type>[optional scope]: <description>
```

and contain the following structural elements:

- fix: a commit that patches a bug in your codebase (this correlates with PATCH in semantic versioning),
- feat: a commit that introduces a new feature to the codebase (this correlates with MINOR in semantic versioning),
- BREAKING CHANGE: a commit that has a footer `BREAKING CHANGE:` introduces a breaking API change (correlating with MAJOR in semantic versioning). A BREAKING CHANGE can be part of commits of any type,
- types other than `fix:` and `feat:` are allowed, for example `build:`, `chore:`, `ci:`, `docs:`, `style:`, `refactor:`, `perf:`, `test:`, and others,
- footers other than `BREAKING CHANGE: <description>` may be provided.

### CODE REVIEW

File: test/controllers/beneficial.owner.gov.controller.spec.ts [hunk 1]
```diff
@@ -127,7 +143,44 @@ describe("BENEFICIAL OWNER GOV controller", () => {
...
+      expect(resp.text).toContain(BO_NOC_HEADING);
       expect(resp.text).not.toContain(TRUSTS_NOC_HEADING);
+      expect(resp.text).toContain(FIRM_NOC_HEADING);
+      expect(resp.text).not.toContain(FIRM_NOC_HEADING_NEW);
+      expect(resp.text).not.toContain(TRUST_CONTROL_NOC_HEADING);
+      expect(resp.text).not.toContain(OWNER_OF_LAND_PERSON_NOC_HEADING);
+      expect(resp.text).not.toContain(OWNER_OF_LAND_OTHER_ENITY_NOC_HEADING);
+    });
+
+    test(`renders the ${config.BENEFICIAL_OWNER_GOV_PAGE} pagen atures of control correctly when FEATURE_FLAG_ENABLE_PROPERTY_OR_LAND_OWNER_NOC is active`, async () => {
```
Comment (markpit) [inline test/controllers/beneficial.owner.gov.controller.spec.ts:155]: pagen atures in description

Comment (mwestacott) [reply test/controllers/beneficial.owner.gov.controller.spec.ts]: Sorted.

File: test/controllers/beneficial.owner.gov.controller.spec.ts [hunk 2]
```diff
@@ -127,7 +143,44 @@ describe("BENEFICIAL OWNER GOV controller", () => {
...
+      expect(resp.text).toContain(config.LANDING_PAGE_URL);
+      expect(resp.text).not.toContain(PAGE_TITLE_ERROR);
+      expect(resp.text).toContain(BENEFICIAL_OWNER_GOV_PAGE_HEADING);
+      expect(resp.text).toContain(SAVE_AND_CONTINUE_BUTTON_TEXT);
+      expect(resp.text).toContain(INFORMATION_SHOWN_ON_THE_PUBLIC_REGISTER);
+      expect(resp.text).toContain(SHOW_INFORMATION_ON_PUBLIC_REGISTER);
+      expect(resp.text).toContain(UK_SANCTIONS_DETAILS);
+      expect(resp.text).toContain(YES_SANCTIONS_TEXT_IT);
+      expect(resp.text).toContain(NO_SANCTIONS_TEXT_IT);
+      expect(resp.text).toContain(SANCTIONS_HINT_TEXT_IT);
```
Comment (dlloyd1-ch) [inline test/controllers/beneficial.owner.gov.controller.spec.ts:176]: Do we need to test for these to be in the response? This test is for NoC. Wouldn't other tests cover this?

File: test/controllers/beneficial.owner.gov.controller.spec.ts [hunk 3]
```diff
@@ -152,7 +205,44 @@ describe("BENEFICIAL OWNER GOV controller", () => {
...
```
Comment (dlloyd1-ch) [inline test/controllers/beneficial.owner.gov.controller.spec.ts:238]: Save as above.

File: test/controllers/beneficial.owner.gov.controller.spec.ts (hunk 2 above)
Comment (mwestacott) [reply test/controllers/beneficial.owner.gov.controller.spec.ts:176]: Yes, agree. Focus should be on NOC testing so will remove the irrelevant assertions (they're covered in the test above).

File: test/controllers/beneficial.owner.gov.controller.spec.ts (hunk 3 above)
Comment (mwestacott) [reply test/controllers/beneficial.owner.gov.controller.spec.ts:238]: Same here. Will change.

File: test/controllers/beneficial.owner.gov.controller.spec.ts (hunk 2 above)
Comment (mwestacott) [reply test/controllers/beneficial.owner.gov.controller.spec.ts]: Fixed this.

File: test/controllers/beneficial.owner.gov.controller.spec.ts (hunk 3 above)
Comment (mwestacott) [reply test/controllers/beneficial.owner.gov.controller.spec.ts]: Sorted.

**Review by markpit — APPROVED** [review]

**Review by dlloyd1-ch — APPROVED** [review]

**Pull request merged**